wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java

```diff
@@ -48,6 +51,15 @@ public static void applyProperties(final Object target, final Properties propert
     }
   }
 
+  public static void addConnAttrForClientIdentification(Dialect dialect, Properties props) {
```

Comment:
Name of the function abbreviates some words and not others, very confusing.
Would suggest addConnectionAttributesForClientIdentification()

Reply:
This logic is dialect-specific. For the moment it's required just for Mysql TAZ. I'd suggest to extend Dialect interface with the following method (just a row idea):
void prepareProperties(Properties props)

This method does nothing for all implementations of Dialect interface. RdsMultiAzDbClusterMysqlDailect will be the only one that add extra properties in such method.

prepareProperties() can be executed in DriverConnectionProvider and DataSourceConnectionProvider.

Reply:
Good idea. I would keep this change out of this PR since it'll noticeably modifies existing classes.